WooCommerce/Classes/Authentication/AuthenticatedWebViewModel.swift

```diff
@@ -1,6 +1,12 @@
 import Foundation
 import WebKit
-import WordPressAuthenticator
+
+/// Optional conformance for a `AuthenticatedWebViewModel` implementation to reload a webview asynchronously.
+protocol WebviewReloadable {
+    /// Allows the view model to load a URL in the webview at anytime.
+    /// This is useful when we have custom redirect handling for WordPress.com login in self-hosted sites.
+    var loadWebview: (_ url: URL) -> Void { get set }
```

Comment:
Just curious.

If I understand this correctly, we reload the same URL again after a particular authentication step. 

1. If the requirement is to reload the `initialURL`, do we still have to pass the `url: URL` as a parameter? 
1. If we remove the URL param, maybe we could name this as `reloadWebview`. i.e. Use `reload` to show that we are trying to load the same URL again.

Reply:
good question. I thought about limiting the ability to only reload the initial URL before, but went for the more flexible way. I agree that it's better to implement just what's needed for now. updated in https://github.com/woocommerce/woocommerce-ios/pull/9217/commits/b31fba13727a63f658f5b74e4cc53acce2416fba